### ISSUE TITLE
[Combo]  JAWS doesn’t announce checked state

### ISSUE THREAD

**Issue by jamesuhl** [IgniteUI/igniteui-angular#15843]
## Description
The JAWS screen reader does not distinguish between checked and non-checked list items for the combo box.  (NVDA does)

## Solution 
Appending/removing  “checkbox checked” on the currently checked boxes using invisible 'screen-reader only text' to the aria-label may be the easiest solution. 

## Details/Additional Context
**NVDA** _does announce_ “not selected” on the checkbox options that are not selected, but it does not announce the checkbox’s state when it _is_ checked. This may be by design, however…
**JAWS** does not announce the state _at all._ Users of JAWS are not informed as the selected state as they are navigating up/down the dropdown. This might be because JAWS is not programmed to recognize web component nodes. Adding “checkbox checked” on the aria-label (and simply removing it when the checkbox is unchecked) will resolve the issue in both screen-readers without sounding redundant in NVDA. 

![Image](https://github.com/user-attachments/assets/d6dbf9d9-f325-4e40-a44d-3b1ca1e66cec)

## More Info on JAWS
The aria-checked attribute on the `<igx-checkbox>` _does update correctly_, but the screen reader does not focus on the `<input>` element. The screen readers only announces the aria-label and the aria-selected on the `<igx-combo-item>` presumably because this node has a role=“option”. The screen reader is not programmed to cascade further into the DOM to search for a checkbox within an “option”. 
The JAWS screen reader may not be announcing the widget properly because JAWS is programmed to be “less tolerant” of various DOM sequences in the code. In the code example below, the combobox code has split up the role=”listbox” and the aria-multiselectable=”true” into two different HTML nodes. The other issue is that the `<igx-combo-item>` that is designated as your listbox options  are not the immediate children of your listbox---the `<igx-display-container>`  is the child. I think a role=“presentation” needs to go on this node to make JAWS function properly. Moving the role= “listbox” to the `<div>` under the search box may fix the issue for JAWS. It seems like JAWS needs all the roles to follow a specific parent/child relationship. Any `<div>` containers added (between the parent and child) for cosmetic purposes need to have role=“presentation”  so JAWS can interpret the widget properly. NVDA seems to be programmed to cascade deeper into the DOM to interpret complex widgets. 

![Image](https://github.com/user-attachments/assets/c80f530f-7d7d-4466-99eb-ec75aa3222db)
I believe the screen reader’s problem with the combobox component is that Infragistics is re-using the `<igx-checkbox>` component inside the `<igx-combobox>` component. Re-using code where possible is always ideal, but in this case, the checkboxes are ignored by the screen readers. It’s the HTML equivalent of doing this:
`<select> <option><input type=”checkbox”></option></select>`

**Comment by ddaribo** [comment]
Hi @jamesuhl,

After further investigation, I determined that [this Chromium issue](https://issues.chromium.org/issues/337904204) is logged about "_aria-selected="true" not indicated to assistive technology_". 

Performing a JAWS test over the [Angular ComboBox Example](https://www.infragistics.com/products/ignite-ui-angular/angular/components/combo#angular-combobox-example) in Chrome, Windows 11, acts as you describe, and I assume you are testing with Chrome as well. One additional detail - the selected state is announced when the aria-selected state changes, but not when navigating.

However, in Firefox the `selected` state is properly announced when navigating (with JAWS), as per your expectation. Please, check out [this recording](https://filetransfer.infragistics.com/index.php/s/pNkirbbfLyWZANy). 

Having this in mind, the combo implementation seems correct in this case, and I am afraid that the Chromium issue has to be addressed in order for this to work as expected. 

Regarding the suggestion about appending "selected" to the `aria-label` of the `igx-combo-item`, I believe that having in mind the default browser behavior, where `selected` is announced after all, this would introduce unwanted duplication - in Firefox currently, and in Chromium later once the issue gets fixed. Just as an example, [this recording](https://filetransfer.infragistics.com/index.php/s/zs2W42WegPksqcw) demonstrates the redundant announcement introduced when `selected` is appended to the item's `aria-label`. In this case JAWS seems to over-announce the element's label, selected state and again label, having in mind that the `aria-activedescendant` is also changed when an element is selected.

Regarding NVDA, it does have its own implementation, where the opposite is announed indeed - "not selected", however, this is not concerned with the browser implementation in this case, I believe.

Please, check out the references and let me know what you think.

**Comment by jamesuhl** [comment]
Thank you for researching the issue with "aria-selected="true"  and Chrome.  I was not aware of this browser defect. The suggestion with using the aria-label is a bit "hacky" , and I agree that it would create duplication.

Yes. I mostly test with JAWS and NVDA in Chrome (v.137.0)
